Tests/SKCoreTests/BuildSystemManagerTests.swift

```diff
@@ -342,8 +360,7 @@ final class BuildSystemManagerTests: XCTestCase {
     let a = DocumentURI(string: "bsm:a.swift")
     let b = DocumentURI(string: "bsm:b.swift")
     let c = DocumentURI(string: "bsm:c.swift")
-    let mainFiles = ManualMainFilesProvider()
-    mainFiles.mainFiles = [a: Set([a]), b: Set([b]), c: Set([c])]
+    let mainFiles = ManualMainFilesProvider(mainFiles: [a: Set([a]), b: Set([b]), c: Set([c])])
```

Comment:
I think (haven’t checked) that we don’t need to add `Set` here. We should be able to infer that the elements are a set from the parameter of `ManualMainFilesProvider`. Same in a few other locations.

```suggestion
    let mainFiles = ManualMainFilesProvider(mainFiles: [a: [a], b: [b], c: [c]])
```